mobilecoind/src/service.rs

```diff
@@ -505,6 +506,83 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
         Ok(response)
     }
 
+    fn get_short_address_hash_impl(
+        &mut self,
+        request: api::GetShortAddressHashRequest,
+    ) -> Result<api::GetShortAddressHashResponse, RpcStatus> {
+        let address = PublicAddress::try_from(request.get_public_address())
+            .map_err(|err| rpc_invalid_arg_error("PublicAddress.try_from", err, &self.logger))?;
+
+        let hash = ShortAddressHash::from(&address);
+
+        let mut response = api::GetShortAddressHashResponse::new();
+        response.set_hash(hash.as_ref().to_vec());
+        Ok(response)
+    }
+
+    fn validate_authenticated_sender_memo_impl(
+        &mut self,
+        request: api::ValidateAuthenticatedSenderMemoRequest,
+    ) -> Result<api::ValidateAuthenticatedSenderMemoResponse, RpcStatus> {
+        // Read the utxo proto
+        let utxo = UnspentTxOut::try_from(request.get_utxo())
+            .map_err(|err| rpc_invalid_arg_error("unspent_tx_out.try_from", err, &self.logger))?;
+
+        let memo_payload = MemoPayload::try_from(&utxo.memo_payload[..])
+            .map_err(|err| rpc_invalid_arg_error("memo_payload.try_from", err, &self.logger))?;
+
+        // Read the sender proto
+        let sender = PublicAddress::try_from(request.get_sender())
+            .map_err(|err| rpc_invalid_arg_error("PublicAddress.try_from", err, &self.logger))?;
+
+        // Get MonitorId from from the GRPC request.
+        let monitor_id = MonitorId::try_from(&request.monitor_id)
```

Comment:
not sure it's any better just stood out compared to the others
```suggestion
        let monitor_id = MonitorId::try_from(request.get_monitor_id())
```